m3u8/writer.go

```diff
@@ -790,6 +882,22 @@ func (p *MediaPlaylist) Encode() *bytes.Buffer {
 			p.buf.WriteString(seg.ProgramDateTime.Format(DATETIME))
 			p.buf.WriteRune('\n')
 		}
+		// handle completed partial segments
+		if p.HasPartialSegments() {
```

Comment:
My understanding was that we should typically keep the partial segments for the last three segments. To me, this looks as if you remove them for all full segments. I think that is too aggressive, but I may be wrong. What's your take on keeping them for three segments?

Reply:
Fine after discussion. Let's keep a separate list, but clean the partial segments as part of the slide operation.